cpp/src/Ice/Initialize.cpp

```diff
@@ -139,83 +139,35 @@ Ice::stringSeqToArgs(const StringSeq& args, int& argc, const wchar_t* argv[])
 #endif
 
 Ice::CommunicatorPtr
-Ice::initialize(int& argc, const char* argv[], InitializationData initData)
+Ice::initialize(InitializationData initData)
 {
-    initData.properties = createProperties(argc, argv, initData.properties);
+    // We clone the properties: if two communicators share the same initData, we don't want them to share the same
+    // properties object.
+    if (initData.properties)
+    {
+        initData.properties = initData.properties->clone();
+    }
 
     CommunicatorPtr communicator = Communicator::create(std::move(initData));
-    communicator->finishSetup(argc, argv);
+    communicator->finishSetup();
     return communicator;
 }
 
 Ice::CommunicatorPtr
-Ice::initialize(int& argc, const char* argv[], string_view configFile)
+Ice::initialize(int& argc, const char* argv[])
 {
-    InitializationData initData;
-    initData.properties = createProperties();
-    initData.properties->load(configFile);
-    return initialize(argc, argv, std::move(initData));
+    InitializationData initData{.properties = createProperties(argc, argv)};
+    return initialize(std::move(initData));
 }
 
 #ifdef _WIN32
 Ice::CommunicatorPtr
-Ice::initialize(int& argc, const wchar_t* argv[], InitializationData initData)
-{
-    Ice::StringSeq args = argsToStringSeq(argc, argv);
-    CommunicatorPtr communicator = initialize(args, std::move(initData));
-    stringSeqToArgs(args, argc, argv);
-    return communicator;
-}
-
-Ice::CommunicatorPtr
-Ice::initialize(int& argc, const wchar_t* argv[], string_view configFile)
-{
-    InitializationData initData;
-    initData.properties = createProperties();
-    initData.properties->load(configFile);
-    return initialize(argc, argv, std::move(initData));
-}
-#endif
-
-Ice::CommunicatorPtr
-Ice::initialize(StringSeq& args, InitializationData initData)
-{
-    IceInternal::ArgVector av(args);
-    CommunicatorPtr communicator = initialize(av.argc, av.argv, std::move(initData));
-    args = argsToStringSeq(av.argc, av.argv);
-    return communicator;
-}
-
-Ice::CommunicatorPtr
-Ice::initialize(StringSeq& args, string_view configFile)
-{
-    InitializationData initData;
-    initData.properties = createProperties();
-    initData.properties->load(configFile);
-    return initialize(args, std::move(initData));
-}
-
-Ice::CommunicatorPtr
-Ice::initialize(InitializationData initData)
```

Comment:
With this initialize, we were NOT cloning the initData.properties. 

However, I think this behavior is unexpected, that's why I changed it in the new initialize(InitializationData).

Reply:
Why is it a problem?

If you look at existing code, we never assume the initData.properties object becomes the properties object of the communicator. We always use "getProperties" on the communicator after initialization. And getProperties in the scripting languages creates a new wrapper.